consumer/pact-jvm-consumer/src/main/java/au/com/dius/pact/consumer/dsl/PactDslJsonArray.java

```diff
@@ -1232,8 +1232,8 @@ public PactDslJsonArray eachArrayWithMinMaxLike(int numberExamples, Integer minS
    * @param example Example value to be used in the consumer test
    */
   public PactDslJsonArray valueFromProviderState(String expression, Object example) {
-    generators.addGenerator(Category.BODY, rootPath + appendArrayIndex(0), new ProviderStateGenerator(expression));
     body.put(example);
+    generators.addGenerator(Category.BODY, rootPath + appendArrayIndex(0), new ProviderStateGenerator(expression));
```

Comment:
Another option here would be to change `appendArrayIndex(0)` to `appendArrayIndex(1)`.  I went the proposed approach instead as it matched most other patterns in the class and it didn't _appear_ there was any dependencies on the order of operations here.